File: test/leetcode/m151/ReverseWordsInAStringTest.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m151;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * Test for {@link ReverseWordsInAString}
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/09/24 10:31
 */
class ReverseWordsInAStringTest {
    private ReverseWordsInAString test;

    @BeforeEach
    void setUp() {
        test = new ReverseWordsInAString();
    }

    @Test
    void test1() {
        String s = "the sky is blue";
        String expected = "blue is sky the";
        assertEquals(expected, test.reverseWords(s));
    }

    @Test
    void test2() {
        String s = "  hello world  ";
        String expected = "world hello";
        assertEquals(expected, test.reverseWords(s));
    }

    @Test
    void test3() {
        String s = "a good   example";
        String expected = "example good a";
        assertEquals(expected, test.reverseWords(s));
    }
}

File: src/leetcode/m151/ReverseWordsInAString_test.go
package m151

import "testing"

func Test_reverseWords(t *testing.T) {
	tests := []struct {
		name string
		s    string
		want string
	}{
		{"Test case 1", "the sky is blue", "blue is sky the"},
		{"Test case 2", "  hello world  ", "world hello"},
		{"Test case 3", "a good   example", "example good a"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := reverseWords(tt.s); got != tt.want {
				t.Errorf("reverseWords() = %v, want %v", got, tt.want)
			}
		})
	}
}
